core/trino-main/src/test/java/io/trino/cost/TestGetTableStatisticsOperations.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cost;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.CountingAccessMetadata;
+import io.trino.metadata.MetadataManager;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.testing.LocalQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.trino.sql.planner.LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.transaction.TransactionBuilder.transaction;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true) // counting metadata is a shared mutable state
+public class TestGetTableStatisticsOperations
+{
+    private final LocalQueryRunner queryRunner;
+    private final CountingAccessMetadata metadata;
+
+    public TestGetTableStatisticsOperations()
+    {
+        queryRunner = LocalQueryRunner.builder(testSessionBuilder().build())
```

Comment:
don't initialize test resources before test class is actually being run by the test runner
init should be done in a `@BeforeClass` method

for this, extend from `AbstractTestQueryFramework` and it will do both setup and cleanup for you.

Reply:
Good point. `AbstractTestQueryFramework` is inside `trino-testing` module which `trino-main` doesn't have dependency on. So I used `BasePlanTest`. It also helps close the query runner. Please let me know if it works.

Reply:
`BasePlanTest` is not the right class to extends from (we're not testing plans here, after all).

Move the test to `trino-tests` instead.

Reply:
Updated.